### DISCUSSION TITLE
Istio ingress into linkerd mesh

### DISCUSSION THREAD

**Discussion by pradeepdesh20** [linkerd/linkerd2#12824] [Q&A]
Hello  quick question, we are in process of migrating from Istio to Linkerd mesh. Due to dependency on end clients, we still want to carry on using istio ingress gateway while we migrate to another service proxy like contour. 
We want to mesh istio ingress gateway into linkerd mesh, would this is possible.

**Comment by wmorgan** [comment]
I answered on the Slack as well, but since those messages die after 90 days: Yes, should be possible! Just read through the [Linkerd ingress docs](https://linkerd.io/2.15/tasks/using-ingress/) carefully to understand whether you need to mesh in ingress mode or normal mode.

And then please write a blog post about your migration :)

**Comment by kflynn** [comment]
@pradeepdesh20 It, uh, turns out that this is actually a little more complex than just reading the ingress docs. Sorry about that! If you want to give it a shot, how about reach out to me directly on Slack? I'm @flynn on the Linkerd or CNCF Slacks. 🙂 Thanks!

**Comment by ThomasEwalts** [comment]
@kflynn Did you guys figure this out back then? I'm running into the exact same issue

**Comment by kflynn** [comment]
@ThomasEwalts Check out https://buoyant.io/blog/migrating-from-istio-to-linkerd -- we describe how this was made to work. Let me know how it goes! 🙂

**Reply by ThomasEwalts** [reply]
I was completely fed up with Istio and have dumped Istio in favor of extending the Cilium CNI to also handle the Gateway. 1 question remains; is there a way to incorporate gateway into the Linkerd mesh or will my days in viz dashboard be spend looking sadly at the dangling "unmeshed" in front of my services?

**Reply by ThomasEwalts** [reply]
@kflynn I have spend some time last weekend to try and figure it out, but is it possible that what i try to do is impossible? The situation is as follows: Cilium now handles Gateway API, which creates a "cilium-envoy" daemonset. These calls get routed directly to the service instead of through the Linkerd mesh. I have googled and experimented a lot, but I can't even get a linkerd proxy to be injected into the cilium-envoy pod (due to hostnetwork enabled, I presume). Is it possible to force all traffic from the cilium-envoy pod into the mesh, instead of directly to the service? Or should I not use the cilium gateway api at all and use a different approach?

**Reply by kflynn** [reply]
@ThomasEwalts Gut feel without knowing exactly what your situation is: if you have Cilium handling Gateway API, I feel like you're running Cilium Service Mesh, and trying to run Cilium Service Mesh _and_ Linkerd is a recipe for... uh... nothing good? 😂
